daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -578,11 +578,9 @@ private[lf] final class Compiler(
       case ECallInterface(iface, methodName, e) =>
         SBCallInterface(iface, methodName)(compile(env, e))
       case EToRequiredInterface(requiredIfaceId @ _, requiringIfaceId @ _, body @ _) =>
-        // TODO https://github.com/digital-asset/daml/issues/11978
-        throw CompilationError("EToRequiredInterface is not implemented")
+        compile(env, body)
       case EFromRequiredInterface(requiredIfaceId @ _, requiringIfaceId @ _, body @ _) =>
-        // TODO https://github.com/digital-asset/daml/issues/11978
-        throw CompilationError("EFromRequiredInterface is not implemented")
+        SBFromRequiredInterface(requiredIfaceId)(compile(env, body))
```

Comment:
I think this should be:

```suggestion
        SBFromRequiredInterface(requiringIfaceId)(compile(env, body))
```

since you're going from `requiredIface` to `requiringIface` and you need to check if the value fits the latter.